### PR TITLE
issue/117 - Calendar enhancements

### PR DESCRIPTION
**Pull request by shawnvarghese**
The major addition in this PR is highlighting days on the calendar which contain events, so it is easier for users to identify them rather than clicking each date separately.
I tried to add the feature by which it would be possible to deselect a date so that the user can see events for the whole month, but that doesn't seem to be trivial to implement or even supported by the jQuery datepicker.

### CODE REVIEW

File: census/views.py [hunk 1]
```diff
@@ -160,7 +182,7 @@ def fetch_events_from_db(self, **kwargs):
         search = kwargs.get('search')
         user_auth_status = kwargs.get('user_auth_status')
 
-        if user_auth_status:
+        if not user_auth_status:
```
Comment (shawnvarghese) [inline census/views.py]: I accidentally got this logic the other way around in my last PR. Correcting it here.

File: census/static/datepicker/js/datepicker.js [hunk 2]
```diff
@@ -30,10 +63,39 @@ $(function () {
...
+        success: function (data) {
+            let eventDates = data.dates;
+
+            $('#datepicker').find(`[data-month='`+(month-1)+`']`)
+                .find("a").filter(function () {
+                return eventDates.indexOf($(this).text()) > -1;
+            }).addClass('ui-datepicker-active-events');
+        }
+    });
+}
```
Comment (mikeubell) [inline census/static/datepicker/js/datepicker.js]: Seems to highlight events in the future but not in the past, which is probably ok, but we display the ones in the past on the home page, maybe we should not?

Comment (shawnvarghese) [reply census/static/datepicker/js/datepicker.js]: It should highlight all events in the month, including past events. I tested it again now and it seems to be working correctly. Is the past event showing up on the list in your homepage? Both calls essentially use the same logic so the behavior shouldn't differ.

Comment (mikeubell) [reply census/static/datepicker/js/datepicker.js]: Here is a screen shot.  Some of the past events are highlighted and some are not.  I think that the past events should not be shown on the home page in any event.  We could leave that for a later enhancement. 
Screen Shot 2019-12-22 at 6 11 44 PM

Comment (mikeubell) [reply census/static/datepicker/js/datepicker.js]: If you want me to gather other information, let me know.

Comment (shawnvarghese) [reply census/static/datepicker/js/datepicker.js]: Thanks @mikeubell. You were right, there was an issue in highlighting single digit dates. I've fixed it now. I'll merge it.

**Pull request merged**